src/Quasar/QuasarF/Interpreter/Internal.purs

```diff
@@ -96,54 +96,48 @@ delete ∷ AX.URL → AXF.AffjaxF RequestContent String
 delete u = AXF.affjax (defaultRequest { method = Left DELETE, url = u })
 
 mkFSUrl
-  ∷ ∀ s s' r
-  . RelDir s
-  → AnyPath Abs s'
-  → URI.Query
-  → Free (Coproduct (CF.ConfigF (Config r)) AXFP) String
-mkFSUrl relDir fsPath q = do
-  uri ← URIRef.print <$> mkFSUrl' relDir fsPath q
-  pure uri
-
-mkFSUrl'
-  ∷ ∀ s s' r
-  . RelDir s
-  → AnyPath Abs s'
-  → URI.Query
-  → Free (Coproduct (CF.ConfigF (Config r)) AXFP) URI.URIRef
-mkFSUrl' relDir fsPath = mkUrl' (bimap (baseify (dir "/")) (baseify (file "")) fsPath)
+  ∷ ∀ r
+  . RelDir
+  → AbsPath
+  → URI.QQuery
+  → AjaxM r String
+mkFSUrl relDir fsPath q = mkUrl (bimap baseify baseify fsPath) q 
   where
-    baseify ∷ ∀ b. Path Rel b s → Path Abs b s' → Path Rel b s
-    baseify x p = relDir </> fromMaybe x (p `relativeTo` rootDir)
+    baseify ∷ ∀ b. IsDirOrFile b => Path Abs b → Path Rel b
+    baseify p = relDir </> p `relativeTo` rootDir
 
-mkUrl ∷ ∀ s r. RelPath s → URI.Query → Free (Coproduct (CF.ConfigF (Config r)) AXFP) String
-mkUrl relPath q = URIRef.print <$> mkUrl' relPath q
+mkUrl ∷ ∀ s r. RelPath → URI.QQuery → AjaxM r String
+mkUrl relPath q = URI.qURIRef.print <$> mkUrl' relPath q
 
-mkUrl' ∷ ∀ s r. RelPath s → URI.Query → Free (Coproduct (CF.ConfigF (Config r)) AXFP) URI.URIRef
+mkUrl' ∷ ∀ s r. RelPath → URI.QQuery → AjaxM r URI.QURIRef
 mkUrl' relPath q = do
   { basePath } ← ask
   pure (bimap toURI toRelativeRef basePath)
   where
     toURI { scheme, authority, path } =
       URI.URI
-        (Just scheme)
-        (URI.HierarchicalPart
-          authority
-          (Just (bimap ((path </> _) <<< sandbox) ((path </> _) <<< sandbox) relPath)))
+        scheme
+        (case authority of
+          Nothing ->
+            URI.HierarchicalPartNoAuth 
+              (Just (bimap (path </> _) (path </> _) relPath))
+          Just authority' -> 
+            URI.HierarchicalPartAuth 
+              authority'
+              (Just (bimap (path </> _) (path </> _) relPath))
+        )
```

Comment:
This is a bit hard to read, maybe we could extract it out into another where function and use it like `toHierPart authority (Just (bimap (path </> _) (path </> _) relPath))`